src/translator/text-based-field.ts

```diff
@@ -45,16 +47,20 @@ export namespace TextBasedField {
                 })
                 .join(", ");
 
-            return [
-                {
-                    validationType:
-                        ELEMENT_VALIDATION_TYPES["Text field"][
-                            EElementType.EMAIL
-                        ].validationTypes[0],
-                    validationRule: extractedDomains,
-                    validationErrorMessage: validation[0].errorMessage,
-                },
-            ];
+            const DOMAIN_REGEX = /@[a-zA-Z0-9-]+\.[a-zA-Z]{2,}/;
+            const isDomain = DOMAIN_REGEX.exec(extractedDomains);
+            if (isDomain) {
+                return [
+                    {
+                        validationType:
+                            ELEMENT_VALIDATION_TYPES["Text field"][
+                                EElementType.EMAIL
+                            ].validationTypes[0],
+                        validationRule: extractedDomains,
+                        validationErrorMessage: validation[0].errorMessage,
+                    },
+                ];
+            }
```

Comment:
shouldnt this be in https://github.com/LifeSG/web-form-builder/pull/78? 
there are comments in that PR on this block of code and this change risks conflicting with changes over there

Reply:
i have merged it with the latest changes from CCUBE-1435 :)

Reply:
alright I will wait for CCUBE-1435 to be merged first before reviewing further